amy/emails/models.py

```diff
@@ -159,14 +165,27 @@ class ScheduledEmail(CreatedUpdatedMixin, models.Model):
         verbose_name="Linked template",
     )
 
+    # This generic relation is limited only to single relation, and only to models
+    # defining their PK as numeric.
+    generic_relation_content_type = models.ForeignKey(
+        ContentType,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+    )
+    generic_relation_pk = models.PositiveIntegerField(null=True, blank=True)
+    generic_relation = GenericForeignKey(
+        "generic_relation_content_type", "generic_relation_pk"
+    )
+
```

Comment:
How do you see us making use of this generic object relation? Is this what will help us to e.g. check a Person's email address is up to date when sending an email?

Reply:
For now it's only going to point at the main object, for which the email was created (e.g. a person for "persons_merged" signal, or a membership for "membership quarterly check"). I admit it's a subject to change, as we can only have 1 object in this generic relationship, and sometimes we would like to have more than that (e.g. for instructor not placed in a workshop we could have a person and an event).